artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/aio/AIOSequentialFileFactory.java

```diff
@@ -304,7 +304,7 @@ public void run() {
          try {
             libaioFile.write(position, bytes, buffer, this);
          } catch (IOException e) {
-            callback.onError(-1, e.getMessage());
```

Comment:
@mtaylor we should do on NIO as well..

The difference I see is testcases.... I would keep it for consistency.